components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/resources/branding-preference.yaml

```diff
@@ -198,13 +198,190 @@ paths:
+          description: Successfully created.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CustomTextModel'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        "409":
+          $ref: '#/components/responses/Conflict'
+        '500':
+          $ref: '#/components/responses/ServerError'
+        '501':
+          $ref: '#/components/responses/NotImplemented'
+
+    put:
+      tags:
+        - Custom Text
+      operationId: updateCustomText
+      summary: Update custom text of a tenant.
+      description: |
+        This API provides the capability to update the custom texts for the specified screen & locale.<br>
+          <b>Permission required:</b> <br>
+            * /permission/admin/manage/identity/applicationmgt/update <br>
+          <b>Scope required:</b> <br>
+            * internal_application_mgt_update
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CustomTextModel'
+        description: This represents the custom text content to be updated for the specified screen & locale.
+        required: true
+      responses:
+        '200' :
+          description: Successfully updated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CustomTextModel'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+    delete:
+      tags:
+        - Custom Text
+      operationId: deleteCustomText
+      summary: Deletes custom text.
+      description: |
+        This API provides the capability to delete the custom texts for the specified screen & locale of a tenant.<br>
+          <b>Permission required:</b> <br>
+            * /permission/admin/manage/identity/applicationmgt/update <br>
+          <b>Scope required:</b> <br>
+            * internal_application_mgt_update
+      parameters:
+        - $ref: '#/components/parameters/typeQueryParam'
+        - $ref: '#/components/parameters/nameQueryParam'
+        - $ref: '#/components/parameters/localeQueryParam'
+        - $ref: '#/components/parameters/screenQueryParam'
+      responses:
+        '204' :
+          description: Successfully deleted.
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
```

Comment:
Is this needed? Even for nonexisting resource deletion we return 204 in other APIs due to idempotent law

Reply:
Addressed with https://github.com/wso2/identity-api-server/pull/495/commits/1758c625c422ac94793b83d06c1f7732e44bc9d1